### ISSUE TITLE
manual_map should not expand macros in suggestion

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#6811]
````rust
pub fn main() {
    let x = Some(3);
    let y = match x {
        Some(i) => Some(vec![i]),
        None => None,
    };
}
````

````
warning: manual implementation of `Option::map`
 --> manual_map.rs:3:13
  |
3 |       let y = match x {
  |  _____________^
4 | |         Some(i) => Some(vec![i]),
5 | |         None => None,
6 | |     };
  | |_____^ help: try this: `x.map(|i| <[_]>::into_vec(box [$($x),+]))`
  |
  ````
